wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/PluginFQNParser.java

```diff
@@ -56,7 +66,7 @@
           + VERSION_PATTERN
           + ")";
   private static final Pattern PLUGIN_PATTERN =
-      Pattern.compile("((?<registry>" + REGISTRY_PATTERN + ")/)?(?<id>" + ID_PATTERN + ")");
+      Pattern.compile("((?<url>" + URL_PATTERN + ")(/?)(#?))?(?<id>" + ID_PATTERN + ")?");
```

Comment:
Why do you need groups for `/` and `#`?
Can't it be simplified in the following way `(/?)(#?)` -> `/?#?`?

Reply:
looks like it can